### DISCUSSION TITLE
How to deal with a large amount of data in a "many-steps" workflow?

### DISCUSSION THREAD

**Discussion by GlaIZier** [argoproj/argo-workflows#7368] [Q&A] [answered]
Hi and have a great day! Newbie here. 

Would you please help me with a question?

I'm trying to incorporate Argo Workflows to build a data pipeline. The pipeline starts with ~6Tb of files with a sprawling folder structure located on a local disk of a remote machine where the workflow is going to be executed (at least for now). This is the only machine in a cluster. Then, this data is needed to be cleaned up and processed in many steps using external services and scripts.

I followed the tutorial and learned that an artifactory is the way to deal with inputs and outputs of workflow's steps. I also know that kubernetes allows to use some persistant volume for pods which I think might also be used to store the inputs and outputs of the steps.

Given that I don't want a lot of data be copied or moved around anywhere yet again in the pipeline, what's the best possible option to deal with inputs and outputs? I also wouldn't want to load the initial data to an artifactory provided that I already have it on a disk. 

P.S. I tried to create a Slack account following the link on the main page, but the link "is no longer active". How can I sign up and ask questions there?

Thank you!

**Comment by thegreymatter** [comment, marked as answer]
there are a couple of ways you can achieve this solution. 
a very simple way for this solution is to mount shared volumes, i.e. it basically means all of your pods will have access to that shared volume.  and you can treat it just as a large shared disk. 

here are a couple of examples on how to do that. 

https://github.com/argoproj/argo-workflows/blob/master/examples/volumes-existing.yaml
https://github.com/argoproj/argo-workflows/blob/master/examples/volumes-pvc.yaml

you can then just transfer through artifacts just the metadata of what data you wish to work with currently (usually filenames)., and access the data itself through the volumes.
